packages/veritone-react-common/src/components/OCREngineOutputView/index.js

```diff
@@ -0,0 +1,122 @@
+import React, { Component, Fragment } from 'react';
+import { arrayOf, shape, number, string, func } from 'prop-types';
+import classNames from 'classnames';
+
+import EngineOutputHeader from '../EngineOutputHeader';
+import withMuiThemeProvider from '../../helpers/withMuiThemeProvider';
+import OCRObject from './OCRObject';
+import styles from './styles.scss';
+
+@withMuiThemeProvider
+class OCREngineOutputView extends Component {
+  static propTypes = {
+    data: arrayOf(
+      shape({
+        startTimeMs: number,
+        endTimeMs: number,
+        sourceEngineId: string,
+        sourceEngineName: string,
+        taskId: string,
+        series: arrayOf(
+          shape({
+            startTimeMs: number,
+            stopTimeMs: number,
+            object: shape({
+              text: string
+            })
+          })
+        )
+      })
+    ),
+    engines: arrayOf(
+      shape({
+        sourceEngineId: string,
+        sourceEngineName: string
+      })
+    ),
+    selectedEngineId: string,
+    onEngineChange: func,
+    onExpandClicked: func,
+    onOcrClicked: func,
+    className: string,
+    currentMediaPlayerTime: number
+  };
+
+  static defaultProps = {
+    data: []
+  };
+
+  handleOcrClick = (startTime, stopTime) => evt => {
+    this.props.onOcrClicked && this.props.onOcrClicked(startTime, stopTime);
+  };
+
+  render() {
+    let {
+      data,
+      className,
+      engines,
+      selectedEngineId,
+      onEngineChange,
+      onExpandClicked,
+      currentMediaPlayerTime
+    } = this.props;
+
+    return (
+      <div className={classNames(styles.ocrOutputView, className)}>
+        <EngineOutputHeader
+          title="Text Recognition"
+          engines={engines}
+          selectedEngineId={selectedEngineId}
+          onEngineChange={onEngineChange}
+          onExpandClicked={onExpandClicked}
+        />
+        <div className={styles.ocrContent}>
+          {data.map(dataObject => {
+            return (
+              <Fragment
+                key={
+                  'ocr-object-group-' +
+                  dataObject.sourceEngineId +
+                  dataObject.taskId
+                }
+              >
+                {dataObject.status === 'FETCHING' && (
+                  <div>Display a progress</div>
+                )}
+                {dataObject.series && (
+                  <span>
+                    {dataObject.series.map((ocrObject, i) => {
+                      {
+                        /* TDO: key may not be unique enough */
+                      }
+                      return (
+                        <OCRObject
```

Comment:
See the comment I left on this component, i'm not convinced this needs to be pulled out. If you decide to keep it then you can just pass in just the `ocrObject` rather than specific fields which will enable you to handle the `onClick` more gracefully within the component.